src/clients/http_client/http_client.ts

```diff
@@ -260,6 +260,8 @@ class HttpClient {
       .catch((error) => {
         if (error instanceof ShopifyErrors.ShopifyError) {
           throw error;
+        } else if (typeof error === 'object' && error.name === 'FetchError' && error.type === 'invalid-json') {
```

Comment:
I might extract this complicated if statement into a function 

`isRetriableError(error)`